Examples/Io/Performance/ActsExamples/Io/Performance/VertexPerformanceWriter.cpp

```diff
@@ -488,291 +436,296 @@ ProcessCode VertexPerformanceWriter::writeT(
-      auto& innerRecoQOverPFitted = m_recoQOverPFitted.emplace_back();
-
-      auto& innerTruthPhi = m_truthPhi.emplace_back();
-      auto& innerTruthTheta = m_truthTheta.emplace_back();
-      auto& innerTruthQOverP = m_truthQOverP.emplace_back();
 
-      auto& innerResPhi = m_resPhi.emplace_back();
-      auto& innerResTheta = m_resTheta.emplace_back();
-      auto& innerResQOverP = m_resQOverP.emplace_back();
-
-      auto& innerResPhiFitted = m_resPhiFitted.emplace_back();
-      auto& innerResThetaFitted = m_resThetaFitted.emplace_back();
-      auto& innerResQOverPFitted = m_resQOverPFitted.emplace_back();
-
-      auto& innerMomOverlap = m_momOverlap.emplace_back();
-      auto& innerMomOverlapFitted = m_momOverlapFitted.emplace_back();
+    if (vertexMatchFraction <= m_cfg.vertexMatchThreshold) {
+      ACTS_DEBUG("Match fraction " << vertexMatchFraction
+                                   << " below threshold. Not matching.");
+      continue;
+    }
 
-      auto& innerPullPhi = m_pullPhi.emplace_back();
-      auto& innerPullTheta = m_pullTheta.emplace_back();
-      auto& innerPullQOverP = m_pullQOverP.emplace_back();
+    // Get references to inner vectors where all track variables corresponding
+    // to the current vertex will be saved
+    auto& innerTrkWeight = m_trkWeight.emplace_back();
+
+    auto& innerRecoPhi = m_recoPhi.emplace_back();
+    auto& innerRecoTheta = m_recoTheta.emplace_back();
+    auto& innerRecoQOverP = m_recoQOverP.emplace_back();
+
+    auto& innerRecoPhiFitted = m_recoPhiFitted.emplace_back();
+    auto& innerRecoThetaFitted = m_recoThetaFitted.emplace_back();
+    auto& innerRecoQOverPFitted = m_recoQOverPFitted.emplace_back();
+
+    auto& innerTrkParticleId = m_trkParticleId.emplace_back();
+
+    auto& innerTruthPhi = m_truthPhi.emplace_back();
+    auto& innerTruthTheta = m_truthTheta.emplace_back();
+    auto& innerTruthQOverP = m_truthQOverP.emplace_back();
+
+    auto& innerResPhi = m_resPhi.emplace_back();
+    auto& innerResTheta = m_resTheta.emplace_back();
+    auto& innerResQOverP = m_resQOverP.emplace_back();
+
+    auto& innerResPhiFitted = m_resPhiFitted.emplace_back();
+    auto& innerResThetaFitted = m_resThetaFitted.emplace_back();
+    auto& innerResQOverPFitted = m_resQOverPFitted.emplace_back();
+
+    auto& innerMomOverlap = m_momOverlap.emplace_back();
+    auto& innerMomOverlapFitted = m_momOverlapFitted.emplace_back();
+
+    auto& innerPullPhi = m_pullPhi.emplace_back();
+    auto& innerPullTheta = m_pullTheta.emplace_back();
+    auto& innerPullQOverP = m_pullQOverP.emplace_back();
+
+    auto& innerPullPhiFitted = m_pullPhiFitted.emplace_back();
+    auto& innerPullThetaFitted = m_pullThetaFitted.emplace_back();
+    auto& innerPullQOverPFitted = m_pullQOverPFitted.emplace_back();
+
+    // Helper function for computing the pull
+    auto pull =
+        [this](const Acts::ActsScalar& diff, const Acts::ActsScalar& variance,
```

Comment:
Why do you need this capture here?

Reply:
I think this forwards the logger which is then consumed in the `ACTS_WARNING`